pkg/upload/dsym.go

```diff
@@ -0,0 +1,176 @@
+package upload
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/ios"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/server"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+)
+
+type Dsym struct {
+	VersionName string      `help:"The version of the application."`
+	Scheme      string      `help:"The name of the scheme to use when building the application."`
+	Dev         bool        `help:"Indicates whether the application is a debug or release build"`
+	DsymPath    string      `help:"Path to the dSYM" type:"path"`
+	Plist       string      `help:"Path to the Info.plist file" type:"path"`
+	ProjectRoot string      `help:"path to remove from the beginning of the filenames in the mapping file" type:"path"`
+	Path        utils.Paths `arg:"" name:"path" help:"Path to directory or file to upload" type:"path" default:"."`
+}
+
+func ProcessDsym(
+	apiKey string,
+	versionName string,
+	scheme string,
+	dev bool,
+	dsymPath string,
+	plistPath string,
+	projectRoot string,
+	paths []string,
+	endpoint string,
+	timeout int,
+	retries int,
+	overwrite bool,
+	dryRun bool,
+) error {
+
+	var buildSettings *ios.XcodeBuildSettings
+	var dsyms *[]*ios.DsymFile
+	var plistData *ios.PlistData
+	var uploadOptions map[string]string
+
+	for _, path := range paths {
+		uploadInfo, err := ios.ProcessPathValue(path, projectRoot)
```

Comment:
The `projectRoot` is passed in just to make sure we use it instead of `path` if `path` is a directory. When `--project-root` takes precedence, we let the user know "--project-root flag set, it's value takes precedence and will be used for upload" in case they don't know why the project root was different to what they expected.

Reply:
I'm not quite clear why it's being used later on in the Xcode project file parsing?

Reply:
Yeah this should not be the case which makes sense since the path is going to have the option of containing a "project root" so I have now removed it to reflect this change.